### PR TITLE
Print Foreign keys for each table

### PR DESCRIPTION
**Pull request by mingjianliu**
Print the foreign key in the SCHEMA.MD

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -71,7 +72,12 @@ func Execute() {
 			logger.Fatalf("couldn't convert data to logical indexes: table %s. reason: %s", tableName, err)
 		}
 
-		_, err = markdown.WriteString(format.CreateTableMarkdown(tableName, table.Comment, columns, logicalIndexes))
+		foreignkeyData, err := format.GetForeignKeyDescriptions(ctx, db, tableName)
+		if err != nil {
+			logger.Println("couldn't query database to fetch foreign key data: table %s. reason: %s", tableName, err)
```
Comment (mingjianliu) [inline cmd/cmd.go]: can't Fatalf on this because there are still some concurrency issues rarely happening

File: cmd/cmd.go [hunk 2]
```diff
@@ -71,7 +72,12 @@ func Execute() {
 			logger.Fatalf("couldn't convert data to logical indexes: table %s. reason: %s", tableName, err)
 		}
 
-		_, err = markdown.WriteString(format.CreateTableMarkdown(tableName, table.Comment, columns, logicalIndexes))
+		foreignkeyData, err := format.GetForeignKeyDescriptions(ctx, db, tableName)
```
Comment (ntindall) [inline cmd/cmd.go]:
```suggestion
		foreignKeyData, err := format.GetForeignKeyDescriptions(ctx, db, tableName)
```

File: format/format.go [hunk 3]
```diff
@@ -87,9 +81,32 @@ func CreateTableMarkdown(tableName string, comment string, columns []ColumnDescr
...
+				wrapBackTicks(fmt.Sprintf("%s.%s", fk.ReferencedTableName, fk.ReferencedColumnName)),
+			})
+		}
+		foreignKeyTable.Render()
+	}
+
 	return tableMarkdown.String()
 }
 
+func formatTable(table *tablewriter.Table, header []string) {
```
Comment (ntindall) [inline format/format.go:103]: ty

**Review by ntindall — APPROVED** [review]
🌱 ty!

File: cmd/cmd.go [hunk 4]
```diff
@@ -71,7 +72,12 @@ func Execute() {
...
+		if err != nil {
+			logger.Printf("couldn't query database to fetch foreign key data: table %s. reason: %s", tableName, err)
+		}
+
+		_, err = markdown.WriteString(format.CreateTableMarkdown(tableName, table.Comment, columns, logicalIndexes, foreignkeyData))
```
Comment (ntindall) [inline cmd/cmd.go]:
```suggestion
		_, err = markdown.WriteString(format.CreateTableMarkdown(tableName, table.Comment, columns, logicalIndexes, foreignKeyData))
```

File: cmd/cmd.go [hunk 5]
```diff
@@ -71,7 +72,12 @@ func Execute() {
 			logger.Fatalf("couldn't convert data to logical indexes: table %s. reason: %s", tableName, err)
 		}
 
-		_, err = markdown.WriteString(format.CreateTableMarkdown(tableName, table.Comment, columns, logicalIndexes))
+		foreignKeyData, err := format.GetForeignKeyDescriptions(ctx, db, tableName)
+		if err != nil {
+			logger.Printf("couldn't query database to fetch foreign key data: table %s. reason: %s", tableName, err)
```
Comment (ntindall) [inline cmd/cmd.go]: hmmm. won't this make the output non-deterministic @mingjianliu ? what is going on here?

should we error out hard in this case?

Comment (mingjianliu) [reply cmd/cmd.go]: yeah I figure out what's broken and made a fix for it.
TL:DR is sqlx doesn't correctly understand the reflection of fields, so we need to make it more explicit in sql

Comment (ntindall) [reply cmd/cmd.go]: ty

**Pull request merged**